INGRID/ingrid.py

```diff
@@ -1501,6 +1520,24 @@ def LoadPatches(self, fname: str = '') -> None:
             self.CurrentTopology.SetupPatchMatrix()
             self.CheckPatches()
 
+    def ApplyUpDownSymmetry(self) -> None:
+        if self.settings["grid_settings"]["up_down_symmetry"]:
+            if self.CurrentTopology.config == "LSN":
+                del self.CurrentTopology.patches["C1"]
+                del self.CurrentTopology.patches["C2"]
+                del self.CurrentTopology.patches["D1"]
+                del self.CurrentTopology.patches["D2"]
+            elif self.CurrentTopology.config in ["SF15", "SF75"]:
+                del self.CurrentTopology.patches["C1"]
+                del self.CurrentTopology.patches["C2"]
+                del self.CurrentTopology.patches["C3"]
+                del self.CurrentTopology.patches["D1"]
+                del self.CurrentTopology.patches["D2"]
+                del self.CurrentTopology.patches["D3"]
+            else:
+                #TODO: Implement up/down symmetry for other geometries
+                raise Exception("Up/down symmetry not yet implemented for config " + self.CurrentTopology.config)
```

Comment:
If we ever do a proper refactor of ingrid, we can probably have each topology state which patches to delete. Could help keep things tidy and trickle in future changes